wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/scm/ScmPersonalAccessTokenFetcher.java

```diff
@@ -71,4 +75,21 @@ public boolean isValid(PersonalAccessToken personalAccessToken)
         "No PersonalAccessTokenFetcher configured for " + personalAccessToken.getScmProviderUrl(),
         personalAccessToken.getScmProviderUrl());
   }
+
+  /**
+   * Iterate over the Set<PersonalAccessTokenFetcher> declared in container and sequentially invoke
+   * {@link PersonalAccessTokenFetcher#isValid(PersonalAccessTokenParams)} method. If any of the
+   * fetchers return an scm username, return it. Otherwise, return null.
+   */
+  @Nullable
+  public String isValid(PersonalAccessTokenParams params)
```

Comment:
I am wondering if we can rename these `isValid` functions on `getUserName`

Reply:
done